### DISCUSSION TITLE
Beacon or spotlight wont get updated when position of item in page changes.

### DISCUSSION THREAD

**Discussion by FunnyPhantom** [gilbarbara/react-joyride#760] [Q&A] [answered]
Hi!
Consider the following sandbox:
https://codesandbox.io/s/snowy-breeze-8qggk?file=/src/App.js

as you will see, when a new items gets rendered into view the beacon or the spotlight won't get updated to accommodate layout change. How can I address this in my codebase?

**Comment by gilbarbara** [comment, marked as answer]
hey @FunnyPhantom 

In your codesandbox example, just add:

```
useEffect(() => {
  window.dispatchEvent(new Event('resize'));
}, [flag]);
```
